Header 컴포넌트 일부 구현

elppaaa opened the pull request:
SearchForm 부분 제외 대부분의 Header 컴포넌트 작성하였음.

elppaaa committed:
Merge pull request #7 from cyjo9603/dev

컴포넌트 생성 방식 예제

elppaaa committed:
Merge remote-tracking branch 'upstream/test' into test

elppaaa committed:
[feat] Header 일부 기능 추가

- Header box 구현
- Logo (임시로 delta icon사용)
- Notification, Profile 아이콘 추가 -- Popover 구현 예정.
- SearchIcon 추가
- global.tsx: body{ background-color } 추가

- #3 진행중.

elppaaa committed:
[feat] Header(Profile, Notification) 컴포넌트 생성

- Notification Button 생성
- Notification Popover Table 생성
- Profile 버튼 추가
- Profile Menu 추가 (내 정보, 로그아웃)

#3 진행중

elppaaa committed:
[feat] Logo, Notification 구조 개선

- Logo 컴포넌트 클릭 시, 메인 페이지로 이동하도록 함
- Notification의 interface 값 위치 이동

develop #3

elppaaa committed:
[fix] Fix notification text color

Notification에서 알림 글자들이 하얗게 보인것 수정
fix #3

elppaaa committed:
[feat] Create Sign in, Sign up button

- 로그인 / 회원가입 버튼 컴포넌트 작성
- Header에서 로그인 상태 확인하는 isLogin props 추가

develop #3

elppaaa committed:
[fix] fix isLogin variable

isLogin이 true 일 때 로그인 화면이 표시되는 것 수정
fix #3

elppaaa committed:
[feat] Add modal component

- 모달 컴포넌트 추가
- children에 내용이 들어가도록 함
- Portal을 이용하여 구현
    - 따라서 storybook으로 확인할 수 없음

develop #3

elppaaa reviewed (commented):
@cyjo9603 Modal 을 구현할 때 방법이 다양하게 있었는데, 포탈을 사용한 예제도 많아서 포탈을 적용해서 구현을 했습니다. 그런데 포탈을 사용하지 않았을 때와의 차이를 느끼지 못해서요. 혹시 특장점이 있나요?
[commit](https://github.com/BUBlackPunch/blackpunch-client/pull/9/commits/816e82b90db84407b783455bd1a345acc10dccde)

elppaaa committed:
[fix] Modal width & blur

- modal에서 최소 폭 줄임
- 파이어폭스에서 blur효과를 대신하여 흰색 배경 적용
- storyboard에서의 테스트를 위해 Portal 임시 제거
- Modal 파라미터로 너비를 받을 수 있도록 변경

fix #3

elppaaa committed:
[feat] Add Profile Modal

Modal을 기반으로 한 ProfileModal 컴포넌트 추가

develop #3

cyjo9603 commented on src/components/Header/ProfileModal.tsx:
```diff
@@ -0,0 +1,190 @@
+import theme from '../../theme';
+
+type Props = {
+  posts: Array<Post>;
```
이런 타입들은 `Array<Post>`보단 `Post[]`를 사용하는게 좋습니다

elppaaa replied:
가독성 차이인가요 기능 차이가 있나요?

cyjo9603 replied:
가독성과 기능은 따로 문제가 없는데 웬만하면 저런식으로 통일하는 편(컨벤션 같은 느낌?)입니다. 제네릭문법은 저런곳보다는 커스텀 타입을 받을때 많이 사용합니다.

cyjo9603 commented on src/components/Header/ProfileModal.tsx:
```diff
@@ -0,0 +1,190 @@
+        & > div {
+          display: flex;
+          justify-content: flex-start;
+          align-items: center;
+          margin-right: 0;
+          flex-wrap: nowrap;
+          margin-bottom: 0.25rem;
+          text-overflow: ellipsis;
+
+          & > div {
+            width: 3px;
+            height: 1rem;
+            margin-right: 7px;
+          }
+
+          & > span {
+            display: inline-block;
+            max-width: 260px;
+            font-size: 0.76rem;
+            font-weight: 300;
+
+            color: ${theme.PRIMARY_FONT};
+            line-height: 1rem;
+            overflow: hidden;
+            white-space: nowrap;
+            text-overflow: ellipsis;
+          }
+
+          & > span:last-child {
+            flex: none;
+            margin-left: auto;
+            padding-left: 3px;
+            width: 5rem;
+            font-size: 0.7rem;
+            color: gray;
+            margin-right: 3%;
+            flex-shrink: 0;
+          }
+        }
+      `}
```
스타일 파일이 너무 길어지면 따로 파일을 분리하면 좋을 것 같습니다

elppaaa replied:
확인

cyjo9603 commented on src/components/Header/Search.tsx:
```diff
@@ -0,0 +1,63 @@
+  const [isClick, setBool] = useState<boolean>(false);
+  return (
+    <SearchWrapper isClick={isClick}>
+      <SearchButton onClick={() => setBool(true)} />
```
전달하려는 함수는 따로 빼서 작성하는 것이 좋습니다.
그리고 따로 빼서 작성할 때 `useCallback(() => {}, [])`의 형태로 만들면 좀더 최적화에 좋습니다.

elppaaa replied:
Search 부분 자체가 수정할 부분이 많아서 전체적으로 손봐야할 것 같아요. 해당부분 참고해서 작성할게요 🙇‍♂️

cyjo9603 reviewed (commented):
@elppaaa 저는 아직 포탈을 안사용해봤는데 사용하는 것이 좋을 것 같네요.
내부에 따로 css속성에 `relative`속성을 가진 컴포넌트가 있으면 스타일이 꼬일 수 도 있어서 외부에 모달을 작성하는 것이 좋은 것 같습니다.

++ 일부 수정하면 좋을 것 같은 코드에 리뷰를 달아 놓았습니다.

++ 제가 요즘 일이 너무 많아서 작업을 별로 못했는데 이번주 주말이랑 다음주에 몰아서 하겠습니다 죄송해요 ㅜㅜ

elppaaa committed:
[style] Change Array<T> to T[]

#9

elppaaa committed:
[refactor]  Impove 'ProfileModal' Component


- ProfileModal 컴포넌트를 따로 폴더로 빼냄
- 중복되는 interface는 extends 로 변경
- style은 styled.ts로 이동

#9

elppaaa committed:
[fix] Change 'Search' component

- Search Form 의 구조, 디자인 문제로 변경하였음
- Header 최소 폭 수정

#3

elppaaa committed:
[feat] Add ValidCheck component

- Add 'ERROR' color in theme
- Add ValidCheck Component

#3

elppaaa committed:
[refactor] Improve modal structure

Modal component 개수 증가로 구조 변경
- 모든 모달 컴포넌트는 Modals에 속함
- 기존 Modal은 ModalFrame으로 변경

refactor #3

cyjo9603 committed:
Merge branch 'develop' into develop
